src/themes/amsterdam/overrides/_filter_group.scss

```
@@ -5,6 +5,15 @@
   box-shadow: inset 0 0 0 1px $euiFormBorderColor;
 }
 
+.euiFilterGroup--compressed {
+  border-radius: $euiFormControlCompressedBorderRadius;
+
+  & .euiFilterButton {
+    height: $euiButtonHeightSmall;
```

Because you're prop name is actually trying to be consistent with form sizing, I'd use the form specific variables for everything.

```suggestion
    height: $euiFormControlCompressedHeight;
```